frontend/src/components/modals/settings/SettingsForm.tsx

```diff
@@ -39,9 +39,11 @@ function SettingsForm({
 }: SettingsFormProps) {
   const { t } = useTranslation();
   const { isOpen: isVisible, onOpenChange: onVisibleChange } = useDisclosure();
-  const advancedAlreadyInUse =
+  let advancedAlreadyInUse =
     !!settings.SECURITY_ANALYZER || !!settings.CONFIRMATION_MODE;
-  // TODO: || model is not in the list
+  if (settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)) {
+    advancedAlreadyInUse = true;
+  }
   const [enableAdvanced, setEnableAdvanced] =
     React.useState(advancedAlreadyInUse);
```

Comment:
Prefer using `React.useState` to ensure the component only re-renders when the state changes

```tsx
  // true if any advanced-related setting is already in use
  const [enableAdvanced, setEnableAdvanced] = React.useState(
    !!settings.SECURITY_ANALYZER ||
      !!settings.CONFIRMATION_MODE ||
      (!!settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)),
  );

  React.useEffect(() => {
    setEnableAdvanced(
      !!settings.SECURITY_ANALYZER ||
        !!settings.CONFIRMATION_MODE ||
        (!!settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)),
    );
  }, [settings, models]);
```

Reply:
So I think we explicitly _don't_ want this useEffect--`advancedAlreadyInUse` should only be calculated once, when the component first renders, to decide what the initial view should be (advanced or not). After that the user is in control

Reply:
Also, for `useState`, I never fully understand React's magic here, but I _think_ you're allowed to set an intermediate variable (`advancedAlreadyInUse`) and pass that as the default to `useState`. Or am I mistaken?

Reply:
Thanks for the explanation! I think it's fixed now